File: module-player/src/main/java/com/zhf/player/VideoPlayerViewModel.java
package com.zhf.player;

import com.zhf.base.model.BaseModel;
import com.zhf.base.model.IModelListener;
import com.zhf.base.viewmodel.MvmBaseViewModel;
import com.zhf.common.contract.BaseCustomViewModel;

import java.util.ArrayList;

/**
 * 应用模块:
 * <p>
 * 类描述:
 * <p>
 *
 */
public class VideoPlayerViewModel
    extends MvmBaseViewModel<IVideoPlayerView, VideoPlayerModel>
    implements IModelListener<ArrayList<BaseCustomViewModel>>
{
    @Override
    protected void initModel()
    {
        model = new VideoPlayerModel();
        model.register(this);
    }

    public void loadData(int videoId){
        model.videoId = videoId;
        model.load();
    }
    
    @Override
    public void onLoadFinish(BaseModel model,
                             ArrayList<BaseCustomViewModel> data)
    {
        if (getPageView() != null) {
            if (data != null && data.size() > 0){
                getPageView().onDataLoadFinish(data);
            }else {
                getPageView().showEmpty();
            }
        }
    }
    
    @Override
    public void onLoadFail(BaseModel model, String prompt)
    {
        if (getPageView() != null){
            getPageView().showFailure(prompt);
        }
    }
    
    @Override
    public void detachUi()
    {
        super.detachUi();
        if (model != null)
        {
            model.unRegister(this);
        }
    }
    
}

File: module-player/src/main/java/com/zhf/player/VideoPlayerViewModel.kt
package com.zhf.player

import com.zhf.base.model.BaseModel
import com.zhf.base.model.IModelListener
import com.zhf.base.viewmodel.MvmBaseViewModel
import com.zhf.common.contract.BaseCustomViewModel

/**
 * created by demoless on 2021/1/29
 * description:
 */
class VideoPlayerViewModel: MvmBaseViewModel<IVideoPlayerView, VideoPlayerModel<*>>(),
        IModelListener<ArrayList<BaseCustomViewModel>> {

    override fun initModel(): VideoPlayerModel<*> {
        model = VideoPlayerModel<Any>()
        model.register(this)
        return model
    }

    fun loadData(videoId: Int) {
        model.videoId = videoId
        model.load()
    }

    override fun onLoadFinish(model: BaseModel<*>, data: ArrayList<BaseCustomViewModel>) {
        if (data.isNotEmpty()) {
            pageView?.onDataLoadFinish(data)
        } else {
            pageView?.showEmpty()
        }
    }

    override fun onLoadFail(model: BaseModel<*>, prompt: String) {
        pageView?.showFailure(prompt)
    }

    override fun detachUi() {
        super.detachUi()
        model.unRegister(this)
    }
}